corehq/util/pagination.py

```diff
@@ -291,7 +293,15 @@ def set_iterator_detail(self, key, value):
     def _save_state(self):
         self.state.timestamp = datetime.utcnow()
         state_json = self.state.to_json()
-        self.couch_db.save_doc(state_json)
+        for x in range(5):
+            try:
+                self.couch_db.save_doc(state_json)
+            except ConnectionError as err:
+                if x < 4 and "BadStatusLine(\"''\",)" in repr(err):
+                    time.sleep(x)
```

Comment:
Yeah, I had briefly considered `2 ** x` rather than `x`, although I didn't think it would matter much since the values will be 0, 1, 2, 3. I suppose 1, 2, 4, 8 might be slightly better, and I'll try that if I have problems with this. I wasn't aware that this bad status line is an indication that couch is down. In fact, I wasn't sure what it meant at all, so I thought trying again immediately was possibly a good idea.

Anecdotally, I hot-patched the django-manage machine with this code, and it hasn't crashed since then, so it seems like it may be working.

Reply:
Yeah I mispoke there, I shouldn't have said down. I meant more like in the phrase "hit 'em when they're down"—so like when couch is struggling.

Anyhow, if it's been working well for you and doesn't seem to be causing the errors, then great!